Torch variable lighting

Msucharz opened the pull request:
Added particle effects and flickering lighting to torches

fairly modular, to add to new object just apply light_variable_intensity.cs to the light source you want to flicker, add the particle system from torch_prefab to objects you want to give off embers, maybe move the emitter to the desired location because they don't play well with weirdly scaled objects

Msucharz committed:
added glowing particles to torch

added light-emitting particles to torch held by player

Msucharz committed:
added flicker to torch lights

added changing intensity controlled via script to torches, also fixed a mistake made in previous commit (particle system on wrong object)

Msucharz committed:
fixed the script that controls torch flickering to not use deprecated function anymore

Msucharz committed:
added the particle system to torch_prefab, moved fire particle model to the correct place, and reactivated the pixelization shader

Hrober0 commented on Assets/Scripts/light_variable_intensity.cs:
```diff
@@ -0,0 +1,36 @@
+    void Update()
+    {
+        curr_intensity += Random.Range(change_rate*(-1),change_rate);
+        if (curr_intensity < min_intensity)
```
` curr_intensity = Mathf.Clamp(curr_intensity, min_intensity, max_intensity)`

Hrober0 commented on Assets/Scripts/TorchManager.cs:
```diff
@@ -6,14 +6,18 @@ public class TorchManager : MonoBehaviour
 {
     [SerializeField] private bool isTorchOn = false;
     [SerializeField] private Light TorchLight;
+    [SerializeField] private ParticleSystem fire_particles;
```
using snake case in C# is a war crime
private variables should be in camelCase `fireParticles`.
pls fix `TorchLight` as well to `torchLight`

Hrober0 commented on Assets/Scripts/TorchManager.cs:
```diff
@@ -6,14 +6,18 @@ public class TorchManager : MonoBehaviour
 
     private void Update()
     {
+        fire_particles = GetComponent<ParticleSystem>();
```
you already assigned it in inspektor why you doining it there?
Also if you have to use `GetComponent` in update you made something wrong.

Hrober0 commented on Assets/Scripts/light_variable_intensity.cs:
```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class light_variable_intensity : MonoBehaviour
```
Another war crime!
Types like struct, class, enum should be in Pascal case `LightVariableIntensity `

Hrober0 commented on Assets/Scripts/light_variable_intensity.cs:
```diff
@@ -0,0 +1,36 @@
+
+public class light_variable_intensity : MonoBehaviour
+{
+    public Light light;
+    public int seed;
+    public float start_intensity;
+    private float curr_intensity;
+    public float change_rate;
+    public float min_intensity;
+    public float max_intensity;
+    private System.DateTime seed_init;
```
1. fix naming
2. There have to be no public variables in code,
if u want to expose variable in inspektor use ` [SerializeField] private float minIntensityt;`
if you want to get access from another script to variable create getter `public float MinIntensityt => minIntensityt;`
3. Try to keep SerializeField and privat fields separated

Hrober0 commented on Assets/Scripts/light_variable_intensity.cs:
```diff
@@ -0,0 +1,36 @@
+
+    void Start()
+    {
+        light = GetComponent<Light>();
```
Set light in inspektor not by `GetComponent`

Msucharz committed:
fixed most if not all problems

1. fixed snake_case to camelCase
2. fixed setting particle system in Update() function
3. fixed (i think?) public fields being not according to unity standard
